### PR TITLE
fix: add telemetry events for new qna, luis and speech flows

### LINKED ISSUE #6635: Ensure Creation / Publishing Telemetry is being captured
## Is your feature request related to a problem? Please describe.

Ensure that Creation / Publishing Telemetry is being captured properly.

### PR DESCRIPTION
**Pull request by benbrown**
## Description

* Add telemetry events described [here](https://github.com/microsoft/BotFramework-Composer/issues/6635#issuecomment-816609551)
* Additionally, track each click on the "get started" task list, including the item id as well as its displayed "priority"
* Add telemetry track for each click in the provision dialog, and track which resources are being created

To Do:

- [x] Review/amend publishing and provisioning
- [x] Review/amend create flow

## Task Item

Fixes #6635

### CODE REVIEW

File: Composer/packages/types/src/telemetry.ts [hunk 1]
```diff
@@ -107,11 +108,29 @@ type QnaEvents = {
   AlternateQnAPhraseAdded: undefined;
 };
 
+type ResourcesItem = {
```
Comment (GeoffCoxMSFT) [inline Composer/packages/types/src/telemetry.ts:111]: We can't import ResourcesItem from where it is normally defined? Might be good to have TelemetryResourcesItem as the name if only used to recording telemetry.

File: extensions/azurePublish/src/components/azureProvisionDialog.tsx [hunk 2]
```diff
@@ -337,18 +346,22 @@ export const AzureProvisionDialog: React.FC = () => {
     setPage(page);
```
Comment (GeoffCoxMSFT) [inline extensions/azurePublish/src/components/azureProvisionDialog.tsx:346]: Isn't it great how I added this function so you had a place to shove telemetry? :)

File: extensions/azurePublish/src/components/azureProvisionDialog.tsx [hunk 3]
```diff
@@ -337,18 +346,22 @@ export const AzureProvisionDialog: React.FC = () => {
...
     switch (page) {
       case PageTypes.AddResources:
+        telemetryClient.track('ProvisionAddResources', {});
```
Comment (GeoffCoxMSFT) [inline extensions/azurePublish/src/components/azureProvisionDialog.tsx:349]: Maybe ProvisionAddResourcesPage or ProvisionAddResourcesNavigate to distinguish between an event of actually adding resources?

**Review by GeoffCoxMSFT — DISMISSED** [review]

File: extensions/azurePublish/src/components/azureProvisionDialog.tsx (hunk 3 above)
Comment (tdurnford) [inline extensions/azurePublish/src/components/azureProvisionDialog.tsx:349]: Nit: If the properties are empty, you don't need to pass an empty object.
```suggestion
        telemetryClient.track('ProvisionAddResources');
```

File: extensions/azurePublish/src/components/azureProvisionDialog.tsx [hunk 4]
```diff
@@ -1059,6 +1081,7 @@ export const AzureProvisionDialog: React.FC = () => {
               style={{ margin: '0 4px' }}
               text={formatMessage('Cancel')}
               onClick={() => {
+                telemetryClient.track('ProvisionAddResourcesCancel', {});
```
Comment (tdurnford) [inline extensions/azurePublish/src/components/azureProvisionDialog.tsx:1084]:
```suggestion
                telemetryClient.track('ProvisionAddResourcesCancel');
```

File: extensions/azurePublish/src/components/azureProvisionDialog.tsx [hunk 5]
```diff
@@ -1036,6 +1057,7 @@ export const AzureProvisionDialog: React.FC = () => {
               text={formatMessage('Next')}
               onClick={() => {
                 if (formData.creationType === 'generate') {
+                  telemetryClient.track('ProvisionShowHandoff', {});
```
Comment (tdurnford) [inline extensions/azurePublish/src/components/azureProvisionDialog.tsx:1060]:
```suggestion
                  telemetryClient.track('ProvisionShowHandoff');
```

File: extensions/azurePublish/src/components/azureProvisionDialog.tsx [hunk 6]
```diff
@@ -941,6 +961,7 @@ export const AzureProvisionDialog: React.FC = () => {
               style={{ margin: '0 4px' }}
               text={formatMessage('Cancel')}
               onClick={() => {
+                telemetryClient.track('ProvisionCancel', {});
```
Comment (tdurnford) [inline extensions/azurePublish/src/components/azureProvisionDialog.tsx:964]:
```suggestion
                telemetryClient.track('ProvisionCancel');
```

File: extensions/azurePublish/src/components/azureProvisionDialog.tsx [hunk 7]
```diff
@@ -337,18 +346,22 @@ export const AzureProvisionDialog: React.FC = () => {
...
       case PageTypes.ConfigProvision:
+        telemetryClient.track('ProvisionConfigureResources', {});
         setTitle(DialogTitle.CONFIG_RESOURCES);
         break;
       case PageTypes.EditJson:
+        telemetryClient.track('ProvisionEditJSON', {});
         setTitle(DialogTitle.EDIT);
         break;
       case PageTypes.ReviewResource:
+        telemetryClient.track('ProvisionReviewResources', {});
```
Comment (tdurnford) [inline extensions/azurePublish/src/components/azureProvisionDialog.tsx:364]:
```suggestion
        telemetryClient.track('ProvisionReviewResources');
```

File: extensions/azurePublish/src/components/azureProvisionDialog.tsx [hunk 8]
```diff
@@ -337,18 +346,22 @@ export const AzureProvisionDialog: React.FC = () => {
...
         break;
       case PageTypes.ChooseAction:
         setTitle(DialogTitle.CHOOSE_ACTION);
         break;
       case PageTypes.ConfigProvision:
+        telemetryClient.track('ProvisionConfigureResources', {});
         setTitle(DialogTitle.CONFIG_RESOURCES);
         break;
       case PageTypes.EditJson:
+        telemetryClient.track('ProvisionEditJSON', {});
```
Comment (tdurnford) [inline extensions/azurePublish/src/components/azureProvisionDialog.tsx:360]:
```suggestion
        telemetryClient.track('ProvisionEditJSON');
```

File: extensions/azurePublish/src/components/azureProvisionDialog.tsx [hunk 9]
```diff
@@ -337,18 +346,22 @@ export const AzureProvisionDialog: React.FC = () => {
...
     switch (page) {
       case PageTypes.AddResources:
+        telemetryClient.track('ProvisionAddResources', {});
         setTitle(DialogTitle.ADD_RESOURCES);
         break;
       case PageTypes.ChooseAction:
         setTitle(DialogTitle.CHOOSE_ACTION);
         break;
       case PageTypes.ConfigProvision:
+        telemetryClient.track('ProvisionConfigureResources', {});
```
Comment (tdurnford) [inline extensions/azurePublish/src/components/azureProvisionDialog.tsx:356]:
```suggestion
        telemetryClient.track('ProvisionConfigureResources');
```

File: Composer/packages/types/src/telemetry.ts (hunk 1 above)
Comment (benbrown) [reply Composer/packages/types/src/telemetry.ts:111]: It is defined inside the extension, so outside the boundaries of the types package..

File: extensions/azurePublish/src/components/azureProvisionDialog.tsx (hunk 2 above)
Comment (benbrown) [reply extensions/azurePublish/src/components/azureProvisionDialog.tsx:346]: FOR REALS.

**Review by GeoffCoxMSFT — APPROVED** [review]

**Review by tdurnford — APPROVED** [review]

**Pull request merged**